src/utils/pst.ts

```diff
@@ -0,0 +1,80 @@
+import axios from "axios";
+
+// ArDrive Profit Sharing Community Smart Contract
+const vertoCacheURL =
+  "https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ";
+
+// Calls the ArDrive Community Smart Contract to pull the fee
+export async function getArDriveFee(): Promise<number> {
+  try {
+    const res = (await axios.get(vertoCacheURL)).data;
+
+    const arDriveCommunityFee = res.state.settings.find(
+      (setting: (string | number)[]) =>
+        setting[0].toString().toLowerCase() === "fee"
+    );
+    return arDriveCommunityFee ? arDriveCommunityFee[1] : 15;
+  } catch {
+    return 0.15; // Default fee of 15% if we cannot pull it from the community contract
+  }
+}
+
+// Gets a random ArDrive token holder based off their weight (amount of tokens they hold)
+export async function selectTokenHolder(): Promise<string | undefined> {
+  // Read the ArDrive Smart Contract to get the latest state
+  const res = (await axios.get(vertoCacheURL)).data;
+  const balances = res.state.balances;
+  const vault = res.state.vault;
+
+  // Get the total number of token holders
+  let total = 0;
+  for (const addr of Object.keys(balances)) {
+    total += balances[addr];
+  }
+
+  // Check for how many tokens the user has staked/vaulted
+  for (const addr of Object.keys(vault)) {
+    if (!vault[addr].length) continue;
+
+    const vaultBalance = vault[addr]
+      .map((a: { balance: number; start: number; end: number }) => a.balance)
+      .reduce((a: number, b: number) => a + b, 0);
+
+    total += vaultBalance;
+
+    if (addr in balances) {
+      balances[addr] += vaultBalance;
+    } else {
+      balances[addr] = vaultBalance;
+    }
+  }
+
+  // Create a weighted list of token holders
+  const weighted: { [addr: string]: number } = {};
+  for (const addr of Object.keys(balances)) {
+    weighted[addr] = balances[addr] / total;
+  }
+  // Get a random holder based off of the weighted list of holders
+  const randomHolder = weightedRandom(weighted);
+  return randomHolder;
+}
+export function weightedRandom(
```

Comment:
Add white space and documentation

Reply:
[Resolved](https://github.com/th8ta/ArConnect/pull/33/commits/66b11719315a0ac4ea310ba6d165889efd900a59)